src/device.ts

```diff
@@ -252,15 +252,25 @@ export class USBDevice {
         return new Promise((resolve, reject) => {
             if (!this.connected) return reject(new Error("close error: device not found"));
             if (!this.opened) return resolve();
-
-            adapter.close(this._handle)
-            .then(resolve)
-            .catch(error => {
-                reject(new Error(`close error: ${error}`));
+            const releaseInterfacePromises = [];
+
+            for (const iface of this.configuration.interfaces) {
+                releaseInterfacePromises.push(this.releaseInterface(iface.interfaceNumber));
+            }
+
+            Promise.all(releaseInterfacePromises)
+            .catch(error => { /* Ignore */ })
```

Comment:
Prefer a continuing `then` rather than a `finally`. I believe the previous catch will only catch errors up to that point:

```typescript 
Promise.all(releaseInterfacePromises)
.catch(error => { /* Ignore */ })
.then(() =-> ...
```